python/dgl/mock_sparse/sddmm.py

```diff
@@ -0,0 +1,57 @@
+"""Sampled Dense-Dense Matrix Multiplication (SDDMM) operator module."""
+import torch
+
+from .sp_matrix import SparseMatrix
+
+__all__ = ["sddmm"]
+
+
+def sddmm(
+    A: SparseMatrix, mat1: torch.Tensor, mat2: torch.Tensor
+) -> SparseMatrix:
+    r"""Sampled-Dense-Dense Matrix Multiplication (sddmm). `sddmm` multiplies
+    two dense matrices :attr:`mat1` and :attr:`mat2` at the nonzero locations
+    of sparse matrix :attr:`A`. Values of :attr:`A` is added to the resulting
+    matrix.
+
+    Mathematically `sddmm` is formulated as:
+
+    .. math::
+        out = (mat1 @ mat2) * spy(A) + A
```

Comment:
Can the operator be used to calculate `(mat1 @ mat2) * spy(A)`? i.e., w/o the addition term

Reply:
No. If `A` has values it will add it to the output. Thats how `sampled_addmm` from PyTorch works.